src/test/java/com/systelab/seed/allergy/AllergyResourceTest.java

```diff
@@ -8,59 +8,66 @@
         return allergy;
     }
 
-    private Allergy getAllergyData(String name, String signs, String symptoms) {
-        Allergy allergy = new Allergy();
-        allergy.setName(name);
-        allergy.setSigns(signs);
-        allergy.setSymptoms(symptoms);
-        return allergy;
+    private void checkAllergyData(Allergy expected, Allergy actual){
+        TestUtil.checkObjectIsNotNull("Allergy", actual);
+        TestUtil.checkField("Name", expected.getName(), actual.getName());
+        TestUtil.checkField("Sign", expected.getSigns(), actual.getSigns());
+        if (expected.getSymptoms() != null)
+            TestUtil.checkField("Symptoms", expected.getSymptoms(), actual.getSymptoms());
     }
 
     @Description("Create a new allergy with name, sign and symptoms")
     @Test
     public void testCreateAllergy() {
         String expectedName = "Tree pollen";
         String expectedSigns = "Watering eyes";
-        Allergy allergy = getAllergyData(expectedName, expectedSigns);
-        Allergy allergyCreated = given().body(allergy)
-                .when().post("/allergies/allergy")
-                .then().assertThat().statusCode(200)
-                .extract().as(Allergy.class);
-        TestUtil.checkObjectIsNotNull("Allergy", allergyCreated);
-        TestUtil.checkField("Name", expectedName, allergyCreated.getName());
+
+        Allergy allergy = getAllergyData(expectedName, expectedSigns, null);
+        Response response = doCreateAllergy(allergy);
+        Allergy allergyCreated = response.then().extract().as(Allergy.class);
+        int status = response.then().extract().statusCode();
+        TestUtil.checkField("Status Code", 200, status);
+        checkAllergyData(allergy, allergyCreated);
     }
 
-    @Description("Create a new allergy with all the information")
+    @Description("Create a new allergy with all information")
     @Test
-    public void testCreateAllergyWithAllInfo() {
-        String expectedName = "Animal dander";
+    public void testCreateAllergyAllInfo() {
+        String expectedName = "Tree pollen";
         String expectedSigns = "Watering eyes";
         String expectedSymptoms = "Sneezing";
-
         Allergy allergy = getAllergyData(expectedName, expectedSigns, expectedSymptoms);
-        Allergy allergyCreated = given().body(allergy)
-                .when().post("/allergies/allergy")
-                .then().assertThat().statusCode(200)
-                .extract().as(Allergy.class);
-        TestUtil.checkObjectIsNotNull("Allergy", allergyCreated);
-        TestUtil.checkField("Name", expectedName, allergyCreated.getName());
+        Response response = doCreateAllergy(allergy);
+        Allergy allergyCreated = response.then().extract().as(Allergy.class);
+        int status = response.then().extract().statusCode();
+        TestUtil.checkField("Status Code", 200, status);
+        checkAllergyData(allergy, allergyCreated);
+    }
+
+    private Response doCreateAllergy(Allergy allergy){
```

Comment:
Move this auxiliar method first for clarity purpose.
The intent is to cover all the endpoints for the CRUD (GET, PUT, POST DELETE)